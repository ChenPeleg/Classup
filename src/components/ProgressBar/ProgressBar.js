import React, { Component } from "react";
import "./ProgressBar.scss";
import AdvanceContext from "../../context/advance-context";
import ProgressNumbers from "./ProgressNumbers/ProgressNumbers";
class ProgressBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      q_number: "2",
      widthOfLine: "0px",
    };
    this.progBarRef = React.createRef();
    this.lineWidthHandler = this.lineWidthHandler.bind(this);
  }
  lineWidthHandler(length_) {
    if (!this.progBarRef.current) {
      setTimeout(() => {
        this.lineWidthHandler(length_);
      }, 500);
      return;
    }
    const startPosition = this.progBarRef.current.getBoundingClientRect().left;
    const lineLength = length_ - startPosition + 20;
    this.setState({ ...this.state, widthOfLine: lineLength + "px" });
  }

  render() {
    return (
      <div className="progbar" ref={this.progBarRef}>
        <div
          className="prog-line"
          style={{ width: this.state.widthOfLine }}
        ></div>
        <AdvanceContext.Consumer>
          {(context) => (
            <ProgressNumbers
              q_counter={{
                q_current: +context.q_number,
                q_total: +context.total_q,
                q_next: +context.q_next,
              }}
              lineWidthHandler={this.lineWidthHandler}
              viewHandler={this.props.viewHandler}
              next_unanswered_q={+context.q_next}
            />
          )}
        </AdvanceContext.Consumer>
      </div>
    );
  }
}

export default ProgressBar;
