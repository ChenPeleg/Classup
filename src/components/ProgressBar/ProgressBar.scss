@import "../../assets/styles/base";
.progbar {
  position: relative;
  padding: 3px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  background-color: $ProgBar_BGrColor;
  width: 100%;
  height: 30px;
}
.prog-line {
  position: absolute;
  left: 0px;
  z-index: 1;
  background-color: $ProgBar_LineColor;
  top: 35%;
  height: 30%;
  width: 100%;
  border-radius: 0px 10px 10px 0px;
  transition: all ease-in-out $Transition_Duration;
}
