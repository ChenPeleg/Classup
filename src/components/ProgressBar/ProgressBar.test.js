import React from "react";
import ProgressBar from "./ProgressBar";
import { shallow } from "enzyme";

const progbar = shallow(<ProgressBar />);

describe("<ProgressBar>", () => {
  it("renders correctly", () => {
    expect(progbar).toMatchSnapshot();
  });
});
