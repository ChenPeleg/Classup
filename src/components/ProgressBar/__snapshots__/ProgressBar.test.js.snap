// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar> renders correctly 1`] = `
<div
  className="progbar"
>
  <div
    className="prog-line"
    style={
      Object {
        "width": "0px",
      }
    }
  />
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</div>
`;
